src/main/java/edu/wpi/grip/core/OutputSocket.java

```diff
@@ -0,0 +1,121 @@
+package edu.wpi.grip.core;
+
+
+import com.google.common.base.MoreObjects;
+import com.google.common.eventbus.EventBus;
+import edu.wpi.grip.core.events.SocketPreviewChangedEvent;
+import edu.wpi.grip.core.events.SocketPublishedEvent;
+
+/**
+ * Represents the output of an {@link Operation}. The OutputSocket also provides the ability to
+ * make the value in a socket published.
+ * @param <T> The type of the value that this socket stores.
+ */
+public class OutputSocket<T> extends Socket<T> {
+    private final EventBus eventBus;
+
+    /**
+     * Indicates if the socket is being previewed
+     */
+    private boolean previewed = false;
+    /**
+     * Indicates if this socket is published or not.
+     */
+    private boolean published = false;
+
+    /**
+     * @param eventBus   The Guava {@link EventBus} used by the application.
+     * @param socketHint See {@link #getSocketHint()}
+     * @param value      See {@link #getValue()}
+     */
+    public OutputSocket(EventBus eventBus, SocketHint<T> socketHint, T value) {
+        super(eventBus, socketHint, value, Direction.OUTPUT);
+        this.eventBus = eventBus;
+    }
+
+    /**
+     * @param eventBus   The Guava {@link EventBus} used by the application.
+     * @param socketHint {@link #getSocketHint}
+     */
+    public OutputSocket(EventBus eventBus, SocketHint<T> socketHint){
+        super(eventBus, socketHint, Direction.OUTPUT);
+        this.eventBus = eventBus;
+    }
+
+    @Override
+    public void setValue(T value){
+        // Check equality first before the value is ste
+        final boolean isNewValue = !this.getValue().equals(value);
+        // Set the value now that we know if there is a new value being stored
+        super.setValue(value);
```

Comment:
`value` and `eventBus` should be proected because the subclasses need to access them.  Your workaround is just a hack to intercept the private variables, it's not better design because you avoid the `protected` keyword.

I would definitely just make the fields  `protected` and avoid all of this.